index.js

```diff
@@ -203,35 +203,49 @@ function createMergeRequest(options){
 					var projectId = project.id;
 					var labels = options.labels || "";
 					var title = "";
-
-					getMergeRequestTitle(options.message).then(function(userMessage){
-						
-						var title = userMessage.split("\n")[0];
-						var description = userMessage.split("\n").slice(2).join("    \n")
-
-						var mergeRequestURL = gitlabURL + "/api/v3/projects/" + projectId + "/merge_requests";
-						
-						gitlab.projects.post("projects/" + projectId + "/merge_requests", {
-							id: projectId,
-							source_branch: sourceBranch,
-							target_branch: targetBranch,
-							title: title,
-							description : description,
-							labels : labels
-						}, function(err, response, body){
-							var mergeRequestResponse = response.body;
-							if(mergeRequestResponse.iid){
-								open(gitlabURL + "/" + projectName + "/merge_requests/" + mergeRequestResponse.iid + (!!options.edit? '/edit' : '') );
-							}
-							if(mergeRequestResponse.message){
-								console.error(colors.red("Couldn't create pull request"));
-								console.log(colors.red(mergeRequestResponse.message.join()));
+					getRemoteForBranch(targetBranch).then(function(targetRemote){
+						getURLOfRemote(targetRemote).then(function(targetRemoteUrl){
+							var targetMatch = targetRemoteUrl.match(regexParseProjectName);
+							if(targetMatch){
+								var targetProjectName = targetMatch[1];
+							} else {
+								console.error(colors.red('Remote at which ' + targetBranch + " is tracked, It's URL doesn't seem to end with .git . It is assumed that your remote URL will end with .git in this utility. ") );
+								console.log('Please contact developer if this is a valid gitlab repository.');
+								process.exit(1);
 							}
+							gitlab.projects.show(targetProjectName, function(targetProject){
+								var targetProjectId = targetProject.id;
+
+								getMergeRequestTitle(options.message).then(function(userMessage){
+
+									var title = userMessage.split("\n")[0];
+									var description = userMessage.split("\n").slice(2).join("    \n")
+
+									var mergeRequestURL = gitlabURL + "/api/v3/projects/" + projectId + "/merge_requests";
+									gitlab.projects.post("projects/" + projectId + "/merge_requests", {
+										id: projectId,
+										source_branch: sourceBranch,
+										target_branch: targetBranch,
+										title: title,
+										description : description,
+										labels : labels,
+										remove_source_branch: true,
```

Comment:
@beckje01 `remove_source_branch ` I am assuming you have added this as a option for yourself. I have no problem in adding this, lets make this configurable or remove it for now.